File: C#/AnaliseDeSentimento/AnaliseDeSentimento/Entidades/Classificacao.cs
using AnaliseDeSentimento.Constantes;
using System.Collections.Generic;

namespace AnaliseDeSentimento.Entidades
{
    public class Classificacao
    {
        public IReadOnlyList<Restaurante> ClassificadosCorretamente { get; private set; }
        public IReadOnlyList<Restaurante> ClassificadosIncorretamente { get; private set; }
        public decimal Acuracia { get; private set; }

        public Classificacao(List<Restaurante> listaRestaurantes)
        {
            MedirAcuraciaClassificacao(listaRestaurantes);
        }

        void MedirAcuraciaClassificacao(List<Restaurante> listaRestaurantes)
        {
            List<Restaurante> classificadosCorretamente = new();
            List<Restaurante> classificadosIncorretamente = new();

            listaRestaurantes.ForEach(restaurante =>
            {
                (ClassificadoCorretamente(restaurante) 
                    ? classificadosCorretamente
                    : classificadosIncorretamente
                ).Add(restaurante);
            });

            ClassificadosCorretamente = classificadosCorretamente;
            ClassificadosIncorretamente = classificadosIncorretamente;

            Acuracia = (decimal)ClassificadosCorretamente.Count / (ClassificadosCorretamente.Count + ClassificadosIncorretamente.Count);
        }

        bool ClassificadoCorretamente(Restaurante restaurante)
            => (restaurante.Classificacao) switch
            {
                EnumClassificacao.Positiva => ClassificacaoPositiva(restaurante.Estrelas),
                EnumClassificacao.Neutra => ClassificacaoNeutra(restaurante.Estrelas),
                EnumClassificacao.Negativa => ClassificacaoNegativa(restaurante.Estrelas),
                _ => false
            };

        bool ClassificacaoNegativa(decimal estrelas)
            => estrelas >= (decimal)1 && estrelas < (decimal)2.5;

        bool ClassificacaoNeutra(decimal estrelas)
            => estrelas >= (decimal)2.5 && estrelas < (decimal)3.5;

        bool ClassificacaoPositiva(decimal estrelas)
            => estrelas >= (decimal)3.5 && estrelas <= 5;
    }
}

File: Python/Entidades/Classificacao.py
from Entidades.Restaurante import Restaurante
from Constantes.EnumClassificacao import EnumClassificacao

class Classificacao:

  def __init__(self, listaRestaurantes):
    self.ClassificadosCorretamente = []
    self.ClassificadosIncorretamente = []
    self.Acuracia = 0
    self.MedirAcuraciaClassificacao(listaRestaurantes)

  def MedirAcuraciaClassificacao(self, listaRestaurantes):
    for restaurante in listaRestaurantes:
      if self.ClassificacaoCorreta(restaurante):
        self.ClassificadosCorretamente.append(restaurante)
      else:
        self.ClassificadosIncorretamente.append(restaurante)

    self.Acuracia = len(self.ClassificadosCorretamente) / (len(self.ClassificadosCorretamente) + len(self.ClassificadosIncorretamente))

  def ClassificacaoCorreta(self, restaurante: Restaurante):
    if restaurante.Classificacao is EnumClassificacao.Positiva:
      return self.ClassificacaoPositiva(restaurante.Estrelas)

    if restaurante.Classificacao is EnumClassificacao.Neutra:
      return self.ClassificacaoNeutra(restaurante.Estrelas)
    
    if restaurante.Classificacao is EnumClassificacao.Negativa:
      return self.ClassificacaoNegativa(restaurante.Estrelas)

    return true
  
  def ClassificacaoNegativa(self, estrelas):
    return estrelas >= 1 and estrelas < 2.5

  def ClassificacaoNeutra(self, estrelas):
    return estrelas > 2.5 and estrelas < 3.5
  
  def ClassificacaoPositiva(self, estrelas):
    return estrelas >= 3.5 and estrelas <= 5

